CDCgov/prime-reportstream
# 2978 Add a new sftpcehck CoolTest that checks all the ignore receivers.

The PR is to test each **receiver that has an sftp transport listed in the ignore organization**, call the 
sftpcheck (http:/<environment.url>/api/check?sftpcheck=ignore.<name>)
API endpoint and verify you get back an OK (200) response that we are able to
connect to the SFTP server. If there is any error (NOT 200) return from ignore
receiver, it should prompt the organization failed + error message as well
as fail message for the sftpcheck test.
- This test is able to run in **staging**, **local**, and **test** environments 
   only and not others.
- When the smoke tests are running in sequential mode then
  have this new test run after '**ping**' and before '**end2end**'.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. In local host environment and in **./prime-router directory**
2. Run the command: `./prime test --run sftpcheck`
    User will see a list on **pass** in green and **failed** in red

Screenshot of ./prime test --run sftpscheck command

**SFTP - Happy Day**

<img width="752" alt="Screen Shot 2021-11-11 at 5 37 03 PM" src="https://user-images.githubusercontent.com/20068283/141384460-546d928d-3382-4ef6-bebd-61487e7d608a.png">

**SFTP - Down**

![Screen Shot 2021-11-15 at 10 10 09 AM](https://user-images.githubusercontent.com/20068283/141817364-d59a7122-95b8-4e60-95f2-15991ef9cb2c.png)

**No Setting Database:**

![Screen Shot 2021-11-15 at 10 11 57 AM](https://user-images.githubusercontent.com/20068283/141817551-40e698f9-aba7-4c36-bd60-ffd4b03d47fa.png)

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing
- [x] Added tests: `./prime test --run sftpcheck

Screenshot of the result of ./gradlew testSmoke

<img width="676" alt="Screen Shot 2021-11-11 at 5 53 48 PM" src="https://user-images.githubusercontent.com/20068283/141385349-402bc3ef-05cd-49a0-8d80-e0ee5efeca3b.png">

### Process

## Fixes
- #issue - List GitHub issues this PR fixes
- https://app.zenhub.com/workspaces/prime-reportstream-5ff4833beb3e08001a4cacae/issues/cdcgov/prime-reportstream/2978

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
